src/mlpack/methods/hoeffding_trees/hoeffding_tree_impl.hpp

```diff
@@ -621,12 +513,16 @@ void HoeffdingTree<
          const arma::Row<size_t>& labels,
          const size_t numClasses,
          const bool batchTraining,
-         const double successProbability,
-         const size_t maxSamples,
-         const size_t checkInterval)
+         const std::optional<double> successProbability,
+         const std::optional<size_t> maxSamples,
+         const std::optional<size_t> checkInterval)
 {
-  Train(data, info, labels, numClasses, batchTraining, successProbability,
-      maxSamples, checkInterval, this->minSamples);
+  Train(data, info, labels, numClasses, batchTraining,
+      (successProbability.has_value()) ? successProbability.value() :
+          this->successProbability,
+      (maxSamples.has_value()) ? maxSamples.value() : this->maxSamples,
+      (checkInterval.has_value()) ? checkInterval.value() : this->checkInterval,
+      this->minSamples);
```

Comment:
```suggestion
          this->minSamples);
```
This is minor, but I think it makes it a little easier to read.